lib/install/deps.js

```diff
@@ -600,6 +600,7 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   var binaryMatches = typeof pkg.bin === 'object' && tree.children.some(function (child) {
     if (child.removed) return false
     if (typeof child.package.bin !== 'object') return false
+    if (!pkg.bin || !child.package.bin) return false
```

Comment:
Thanks for the good catch/patch! I think the actual point of the bug is the conditional expression `typeof variable === 'object'` and `typeof variable !== 'object'` . It's sometimes annoy since the variable could be `null` at the case and `typeof null` is `object`. So checking `null` would be better imo, like the following:
```diff
-- var binaryMatches = typeof pkg.bin === 'object' && tree.children.some(function (child) {
++ var binaryMatches = pkg.bin !== null && typeof pkg.bin === 'object' && tree.children.some(function (child) {
     if (child.removed) return false
++   if (child.package.bin === null) return false
     if (typeof child.package.bin !== 'object') return false
       return Object.keys(child.package.bin).some(function (bin) {
         return pkg.bin[bin]
     })
   })
```
Cheers!